cmd/helm/install.go

```diff
@@ -192,6 +193,10 @@ func runInstall(args []string, client *action.Install, valueOpts *values.Options
 		return nil, err
 	}
 
+	if _, err := isLimitSize(chartRequested.Raw); err != nil {
+		return nil, err
+	}
```

Comment:
This size issue is not with the requested chart but rather the release object as it is stored. The release object contains some information in addition to the requested chart and the chart is not stored raw. This is checking the wrong object.

Reply:
> I like the idea of providing a better error message here. I think there is a better way to approach this.
> 
> First, this should be built into both install and upgrade.
> 
> Second, it should happen in the `action` package so that anything using the action will get it.
> 
> Third, it should be able to support different storage mechanisms size constraints.
> 
> I wonder if writing this secret and catching the failure is a good way to do it.



> I like the idea of providing a better error message here. I think there is a better way to approach this.
> 
> First, this should be built into both install and upgrade.
> 
> Second, it should happen in the `action` package so that anything using the action will get it.
> 
> Third, it should be able to support different storage mechanisms size constraints.
> 
> I wonder if writing this secret and catching the failure is a good way to do it.

Thank you for your patient review. I didn't think about storing more when I submitted the code. I tried to add the error prompt where I think it is appropriate. Please give some suggestions. I have tested all four supported storage formats, and only ConfigMaps and Secrets have this limitation. Should this problem be treated as a feature fix in v3.3 ?